wordpress-mobile/WordPress-iOS
# User Profile: show site modal when site row tapped

Ref #16245 

When the site is tapped on the User Profile, the site is displayed using the Reader site view it in a fullscreen modal.

The Reader site view is what is presented when tapping the site name on a Reader post card. The differences here are:
- The fullscreen modal.
- `X` left nav bar button to dismiss the view.

Otherwise they should look the same.
 
| Reader Site | User Site |
|--------|-------|
| <img width="469" alt="reader_site" src="https://user-images.githubusercontent.com/1816888/114767503-f2092080-9d24-11eb-92ce-f8d748ee6e79.png"> | <img width="471" alt="user_site" src="https://user-images.githubusercontent.com/1816888/114767534-f8979800-9d24-11eb-9ec7-693952f43dfc.png"> |

---
To test:

Since we don't have real user info yet, I've stubbed in a bogus siteID. You'll need valid siteIDs for the modal to show. The [siteID](https://github.com/wordpress-mobile/WordPress-iOS/blob/e56b67b9d3d85419685980c57affc8f8fc8d6d13/WordPress/Classes/ViewRelated/User%20Profile%20Sheet/UserProfileSheetViewController.swift#L134) is set in  `UserProfileSheetViewController:showSite`.

---
With an invalid siteID (as-is):
- Enable the `newLikeNotifications` feature.
- Go to Notifications > Likes and tap a notification.
- Tap a user, then the site row.
- Nothing will happen visually. This error will appear in the logs: `Error: could not find ReaderSiteTopic for siteID [xxx]`

---
With a valid siteID:
- Set the siteID value as mentioned above.
- Go to Notifications > Likes and tap a notification.
- Tap a user, then the site row.
- Verify the site is displayed:
  - In a fullscreen modal.
  - With the `X` left nav bar button.


| iPhone | iPad |
|--------|-------|
| ![iphone](https://user-images.githubusercontent.com/1816888/114768131-b6bb2180-9d25-11eb-9a32-a42778bb29b7.png) | ![ipad](https://user-images.githubusercontent.com/1816888/114768142-ba4ea880-9d25-11eb-9cc4-d6a33840574a.png) |

---
## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete.

PR submission checklist:
